### ISSUE TITLE
Unable to authenticate or use Google Keep extension in Brave

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#3650]
## Description 

Google Keep extension installs but doesn't allow users to login to their google account in order to use it. `Sign in` button on extension panel is unresponsive - logging into Keep via G Keep website also has no effect.
![image](https://user-images.githubusercontent.com/3299098/54092569-d283ba80-434a-11e9-9cf1-ffae01b9c497.png)

## Steps to Reproduce

   1. Install [Google Keep extension](https://chrome.google.com/webstore/detail/google-keep-chrome-extens/lpcaedmchfhocbbapmcbpinfpgnhiddi/related?hl=en) 
   2. Visit any website
   3. Click `Sign in` to use extension

## Actual result:

Sign-in button is unresponsive.

## Expected result:
Successfully authenticatate into google account in Keep extension.

## Reproduces how often: 

Always

## Brave version (brave://version info)

Version 0.60.48 Chromium: 72.0.3626.121 (Official Build) (64-bit)

### Reproducible on current release:
- Does it reproduce on brave-browser dev/beta builds? 
Yes

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? No
- Is the issue reproducible on the latest version of Chrome? Yes

### Additional Information

User Reports:
https://community.brave.com/t/google-keep-application-cannot-sign-in/34972/14?preview_theme_id=24

**Comment by dinoboy197** [comment]
I also have a very similar issue. I'm using Brave: 0.61.51 Chromium: 73.0.3683.75 (Official Build) (64-bit) on Linux.

I use Keep all day, so having it be disabled is an unfortunate impediment to me being able to use the Brave browser exclusively.

**Comment by rebron** [comment]
Might be fixed with the social networking landing in 64.x. Let's test this out again when that fix lands.

**Comment by jawadalkassim** [comment]
same issue on Version 0.61.51 Chromium: 73.0.3683.75 (Official Build) (64-bit)
Windows 10 1809

**Comment by dinoboy197** [comment]
Thanks @rebron ! Will watch this issue for updates.

**Comment by Brave-Matt** [comment]
+1 from Community:
https://community.brave.com/t/unable-to-sign-in-to-google-keep-extension/48661/4

**Comment by theNailz** [comment]
(Still) broken on `Version 0.64.77 Chromium: 74.0.3729.169 (Official Build) (64-bit)`.

**Comment by tayfurunal** [comment]
still broken on Version 0.65.118 Chromium: 75.0.3770.80 (Official Build) (64-bit)

**Comment by kwroebuck** [comment]
still broken on Version 0.65.121 Chromium: 75.0.3770.100 (Official Build) (64-bit)

**Comment by Brave-Matt** [comment]
Another +1 from Community:
https://community.brave.com/t/unable-to-sign-in-to-google-keep-extension/48661/5?u=mattches

**Comment by coleberhorst** [comment]
still broken on Version 0.66.100 Chromium: 75.0.3770.142 (Official Build) (64-bit)

**Comment by slabo** [comment]
still same issue Version 0.65.121 Chromium: 75.0.3770.100 (Official Build) (64-bit), on MX Linux, Linux mexer 4.19.0-1-amd64 #1 SMP Debian 4.19.5-2~mx17+1 (2018-12-12) x86_64 GNU/Linux

**Comment by ante316** [comment]
**+1**; still an issue.
**OS**: macOS
**Brave**: Version 0.67.123 Chromium: 76.0.3809.87 (Official Build) (64-bit)

This saddens me greatly :(

**Comment by idontcar3** [comment]
+1; still an issue.
OS:Debian 10 (Buster)
Brave: Version 0.68.131 Chromium: 76.0.3809.100 (Official Build) (64-bit)

I was just getting the hinge of google keep and wanted to have it on the desktop.

I{ll find something for:expressionless:  KDE.

**Comment by Brave-Matt** [comment]
+1 reddit
https://www.reddit.com/r/brave_browser/comments/d9o94b/cant_log_into_extensions_with_google_acct_oauth/

**Comment by Brave-Matt** [comment]
+1 from Commiunity:
https://community.brave.com/t/google-keep-application-cannot-sign-in/34972/17?u=mattches

**Comment by brimwats** [comment]
+1 from me -- windows 10

**Comment by ansaris** [comment]
+1
It's not specific to Google Keep, it happens when using Google Sign-in oauth on any 3rd party site. 

Fix for me: Enabling all cookies, then clearing all history/cache/cookies. Attempt to log in again.
I had already disabled shields on both the 3rd party site and the Google signin popup window.
Then I changed it back to disabling only cross-site cookies. Still working until the next time I need to login.
Using Version 0.69.135 Chromium: 77.0.3865.120 (Official Build) (64-bit) on Windows 10.

**Comment by ansaris** [comment]
Please merge with **#6408 Login with Google OAuth Not Working**

**Comment by bsclifton** [comment]
cc: @jumde who is looking at the OAuth issue...

**Comment by jumde** [comment]
assigned it to myself.

**Comment by dlorde** [comment]
Any progress on this?

**Comment by juanjosezg** [comment]
They only ignore this issue

**Comment by coleberhorst** [comment]
Largely this issue and a lot of other polish issues is why I stopped using Brave. Would love to use Brave, but I can't sacrifice productivity extensions just for privacy principles.

**Comment by ryanbr** [comment]
Nothing is showing up in the console when trying to sign in or add to Google keep.

**Extension source:**
  `!DOCTYPE html><html><head><title>Google Keep Chrome Extension</title><script type="text/javascript" src="keep_ba-prodbootstrap.js"></script><link rel="stylesheet" href="open_sans.css"><link rel="stylesheet" href="roboto.css"><link rel="stylesheet" href="roboto_condensed.css"><link rel="stylesheet" href="roboto_slab.css"></head><body><div id="keep-it-container"></div></body></html>`

**keep_ba-prodbootstrap.js:**
https://pastebin.com/raw/MvPdUNkU

**Comment by ryanbr** [comment]
Using Google Drive extension does generate some errors when attempting to login.

![google-drive-ext](https://user-images.githubusercontent.com/1659004/72185268-486b1d80-3457-11ea-824b-b5dbe5d78d59.png)

https://pastebin.com/nW2cshk1

**Comment by aarwdc** [comment]
I have also been unable to sign into keep app in Brave or Brave Beta. Haven't read all the comments to see if this was posted, you can still get to your Keep notes, and add new ones by simply going to https://keep.google.com/. Its all there without the app, though if you want to save a link to a particular page, you have to actually copy and paste it in your note.

**Comment by ianmf6** [comment]
I am having the same issues with Google Keep, Google Calendar, Save to Google Drive browser extensions. Sign in to Google using Keep extensions doesn't do anything. Google Calendar gets stuck in authorizing account.

**Comment by Sneakyghost** [comment]
+1 not working here either, windows 10 pro 1909 running brave Version 1.2.43 Chromium: 79.0.3945.130 (Official Build) (64-bit)

Tried this: disabled all Brave Shield features, additional security extensions such as uBlock Origin, Ghostery, then disabled my PiHole DNS as well and then PeerBlock. Even removed ESET  Endpoint AV protection temporarily to see if anything hinders sign in, but all that is not it and I can also say that on another machine, where I keep a copy of Google Chrome to verify stuff against Brave, it works with all my protection mechanisms active.

**Comment by davidgenesiscruz** [comment]
+1
Version: 1.2.42 Chromium: 79.0.3945.117（Official Build） (64-bit) on macOS

**Comment by awmottaz** [comment]
Not working for me either.
Version 1.5.112 Chromium: 80.0.3987.132 (Official Build) (64-bit)
Using Ubuntu (KDE Neon)

**Comment by panchalkalpesh** [comment]
+1
Not working for me either. 

Shields UP: 
![Screenshot 2020-03-21 at 11 06 18 PM](https://user-images.githubusercontent.com/1808686/77241462-d0447500-6bc8-11ea-9460-69fc1bb948dc.png)

Tested on:
```
Brave		: 1.5.115 Chromium: 80.0.3987.149 (Official Build) (64-bit)
Revision	: 5f4eb224680e5d7dca88504586e9fd951840cac6-refs/branch-heads/3987_137@{#16}
OS		: macOS Version 10.14.3 (Build 18D109)
JavaScript	: V8 8.0.426.27
Flash		: (Disabled)
User Agent	: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36
```

**Comment by andersmmg** [comment]
+1 for me, on latest

**Comment by SAMDEV-R** [comment]
+1 for me, on the latest

**Comment by Brave-Matt** [comment]
Additional user reports on Community:
https://community.brave.com/t/unable-to-sign-in-to-google-keep-extension/48661/11?u=mattches

@rebron is this something we intend or are able to resolve?

**Comment by sazaog** [comment]
+1 for me, on the latest

**Comment by john-oc** [comment]
+1 for me, on the latest

**Comment by michaelignacio** [comment]
+1, latest version of Brave Browser, Windows 10 machine

**Comment by bonzofenix** [comment]
+!

**Comment by Kwikxilver** [comment]
+1, on latest version of browser and still having issues signing into Keep.

**Comment by jgbrown42** [comment]
+1 - still not fixed in Version 1.10.97

**Comment by bsclifton** [comment]
Thanks for your patience folks - @jumde has a solution but we put that on hold while implementing Sync v2. The new sync implementation also has authentication code (ex: using DICE/GAIA) which needed to be resolved before we can pick this back up. Work in progress at https://github.com/brave/brave-core/pull/5616

**Comment by jumde** [comment]
Work will resume on this shortly, stay tuned!

**Comment by ALEG1ND** [comment]
+1 for me, on the latest

**Comment by ThatsNotAName** [comment]
+1

**Comment by LaurenWags** [comment]
@jumde should this one have `QA/Yes` or `QA/No` label?

**Comment by jumde** [comment]
@LaurenWags - Updated the label and the test plan.

**Comment by LaurenWags** [comment]
thank you @jumde 😄

**Comment by foodfan** [comment]
I'm having problems since this change. I was logged out of all google accounts. When I logged in to gmail and google apps accounts, everything worked fine. But when I tried logging into my university alumni account, which has separate authentication, I was properly redirected, logged in, did 2FA, but when redirected to Google, was asked to "reconfirm it was me", I clicked the button but it said there was an error. Also, google drive sync crashed and logged me out. What to do?

**Comment by GeetaSarvadnya** [comment]
Verification passed on

```
Brave | 1.14.66 Chromium: 85.0.4183.83 (Official Build) dev (64-bit)
-- | --
Revision | 94abc2237ae0c9a4cb5f035431c8adfb94324633-refs/branch-heads/4183@{#1658}
OS | Windows 10 OS Version 1903 (Build 18362.1016)

```
- Verified the test plan from https://github.com/brave/brave-core/pull/5616
-  Verified that login to `google-keep extension` is authenticated by using google account successfully
![image](https://user-images.githubusercontent.com/38657976/91552600-b19d9380-e949-11ea-962b-1127a502d905.png)
- Verified that login to `google-calendar`  is authenticated by using google account successfully
![image](https://user-images.githubusercontent.com/38657976/91553083-7c457580-e94a-11ea-9629-2d4663ee8812.png)
- Logged a follow-up issue https://github.com/brave/brave-browser/issues/11493

**Comment by jumde** [comment]
Thanks for the feedback @foodfan - I'm investigating the issue that you're running into. Will post updates here.

**Comment by bsclifton** [comment]
@foodfan (love the name) - just to confirm, you're on Beta or Nightly? This code hasn't reached Release (1.13) channel yet

**Comment by foodfan** [comment]
> @foodfan (love the name) - just to confirm, you're on Beta or Nightly? This code hasn't reached Release (1.13) channel yet

Haha, thanks. 

I am on the Beta channel. Version 1.14.65 Chromium: 85.0.4183.83 (Official Build) beta (64-bit)

Hope that helps

**Comment by foodfan** [comment]
Seems like this was a cookie problem. To fix, I deleted all "google" cookies and re-logged in to my primary (gmail), google apps, and finally alumni account. Note: I did not log out of the extension (not sure if it's a separate cookie or not) but it continued working. I was also able to authenticate Google Backup & Sync. Everything seems to be working now. Thanks for the responsiveness to this and for making Keep work.

**Comment by jumde** [comment]
Thanks for the update @foodfan , if you run into any additional issues let us know. 👍

**Comment by dsg38** [comment]
In case any one else is stuck at the same point - emphasising part of @GeetaSarvadnya post - it's essential that that `Allow Google login for extensions` has been enabled in settings. As this was blocking it for me and clicking 'sign in' on the keep extension did not prompt this setting request

**Comment by jtnydv** [comment]
Still facing this issue and none of the suggested solutions works.

When I enable `Allow Google login for extensions` I keep getting the Google login screen and the extension doesn't work.
Can anyone suggest any workaround for this?

**Comment by jumde** [comment]
Thanks for the report @jtnydv - I verified that this is a regression. Tracking the issue here: https://github.com/brave/brave-browser/issues/14041 - Will try to resolve this soon.

**Comment by JoseJuanSE** [comment]
I allowed `google login for extensions`, now when I push `sign in` button, I get redirected to google login, after I logged, I'm again in the same place. Google keep is still asking me for log in.

**Comment by nickgroverx** [comment]
I did these two and it worked for me:
1. Allow google login for extensions, and 
2. Allow all cookies.

**Comment by matthughes404** [comment]
After updating to the latest Brave version, the Google Keep extension is working properly for me.

Version 1.23.73 Chromium: 90.0.4430.85 (Official Build) (64-bit)

**Comment by KharmaScribbles** [comment]
Best fix and best extension for Google Loyalists is just use https://apps.jeurissen.co/black-menu-for-google extension, when I installed Brave and asked it to import my extensions from Chrome, all my extensions worked from the get-go and all google apps work within the black-menu-for-google extension.
